pkg/api/handlers/compat/images_build.go

```diff
@@ -573,14 +573,15 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 		ulimits = m
 	}
 
+	pullPolicyMap := buildahDefine.PolicyMap
+	pullPolicyMap["true"] = buildahDefine.PullAlways
+	pullPolicyMap["false"] = buildahDefine.PullNever
 	pullPolicy := buildahDefine.PullIfMissing
 	if utils.IsLibpodRequest(r) {
-		pullPolicy = buildahDefine.PolicyMap[query.PullPolicy]
+		pullPolicy = pullPolicyMap[query.PullPolicy]
 	} else {
 		if _, found := r.URL.Query()["pull"]; found {
-			if query.Pull {
-				pullPolicy = buildahDefine.PullAlways
-			}
+			pullPolicy = pullPolicyMap[query.Pull]
```

Comment:
Both cases do not seem to validate the input at all? If no match is found in the map the pullPolicy will be empty instead of returning error when an invalid value is specified.

Reply:
Added a case to handle this.